Consolidate partial/full facade projects

mellinoe opened the pull request:
Work-in-progress for #4941 .

Will rebase when finished.

EDIT: I have finished everything except for projects that currently have some PR's open. I can do those after the PR's merge.

mellinoe commented:
We need to fix the project references on the TFS side, because some of the project files were renamed (dropped the ".CoreCLR" suffix).

weshaggard commented on src/System.Collections/src/System.Collections.csproj:
```diff
@@ -10,14 +10,13 @@
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-  </PropertyGroup>
-  <ItemGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
```
nit: For consistency can you remove the spaces at the begining and ending of these condition strings.

weshaggard commented on src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj:
```diff
@@ -10,8 +10,8 @@
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
-    <NuGetTargetFrameworkMoniker>DNXCore,Version=v5.0</NuGetTargetFrameworkMoniker>
     <NoWarn>0436</NoWarn>
+    <ResourcesSourceOutputDirectory Condition="'$(TargetGroup)' == 'netcore50aot' Or '$(TargetGroup)' == 'net46'">None</ResourcesSourceOutputDirectory>
```
This condition would probably better as TargetGroup != ''

mellinoe replied:
What is the "default" group called?

weshaggard replied:
Right now the default target group is the empty string.

weshaggard commented on src/System.Diagnostics.Tools/src/System.Diagnostics.Tools.csproj:
```diff
@@ -9,12 +9,17 @@
     <AssemblyName>System.Diagnostics.Tools</AssemblyName>
```
FYI: A lot of these project renames will break our internal builds as they are referenced by our wrapper projects. So you will need to handle that when this is merged.

mellinoe replied:
Yeah, I mentioned that above. I'll have to watch the mirror and have a checkin ready.

weshaggard commented on src/System.Runtime.Extensions/src/System.Runtime.Extensions.csproj:
```diff
@@ -23,14 +23,16 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU' " />
-  <ItemGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Debug|AnyCPU'" />
```
nit: more spaces at the beginning and ending of these conditions. Perhaps we should do a find/replace to clean these up across all the projects. Feel free to do that as a separate change.

weshaggard commented on src/System.Runtime.InteropServices/src/System.Runtime.InteropServices.csproj:
```diff
@@ -7,7 +7,8 @@
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
-    <NuGetTargetFrameworkMoniker>DNXCore,Version=v5.0</NuGetTargetFrameworkMoniker>
+    <!-- Force string resources to be excluded for full facades. -->
+    <ResourcesSourceOutputDirectory Condition="'$(TargetGroup)' == 'netcore50aot' Or '$(Targetgroup)' == 'net46'">None</ResourcesSourceOutputDirectory>
```
I would change this TargetGroup!=''

mellinoe commented:
@weshaggard I addressed your comments (and went through all the files I touched and cleaned up the formatting/spacing) and rebased on top of master. I'm getting a TFS checkin ready for the renames now. I believe the projects renamed are:
* System.Threading.CoreCLR
* System.Private.Uri.CoreCLR
* System.Runtime.Handles.CoreCLR
* System.Runtime.Extension.CoreCLR
* System.Runtime.InteropServices.CoreCLR
* System.Threading.Overlapped.CoreCLR
* System.Reflection.TypeExtensions.CoreCLR
* System.Diagnostics.Tools.CoreCLR

All dropped their ".CoreCLR" suffix.

weshaggard commented on src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj:
```diff
@@ -90,6 +94,11 @@
     </Compile>
   </ItemGroup>
   <ItemGroup>
+    <TargetingPackReference Include="mscorlib" Condition="'$(TargetGroup)' == 'net46'" />
```
Should these conditions be on the ItemGroup to help separate and reduce redundancy.

mellinoe replied:
Would you prefer there be two separate item groups for these reference groups?

weshaggard commented on src/System.Runtime.InteropServices/src/System.Runtime.InteropServices.csproj:
```diff
@@ -32,6 +37,13 @@
     <Compile Include="System\Runtime\InteropServices\ComTypes\tymed.cs" />
   </ItemGroup>
   <ItemGroup>
+    <TargetingPackReference Include="mscorlib" Condition="'$(TargetGroup)' != 'netcore50aot'" />
```
I would also move these conditions to the ItemGroup.

mellinoe replied:
There's three different conditions here, it will be quite messy to have different ItemGroups for these, IMO. How would you organize them?
